File: backend-module/src/main/java/com/campus/eneity/Student.java
package com.campus.eneity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Student {
    private int id;
    private int userId;
    private String studentNo;
    private String name;
    private String gender;
    private int classId;
    private String enrollYear;
    private int status;
    private String phone;
    private String guardianPhone;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private boolean isDeleted;
}

File: frontend/src/types/student.ts
export interface Student {
  id: number
  userId: number
  studentNo: string
  name: string
  gender: string | null
  classId: number
  enrollYear: string | null
  status: number
  phone: string | null
  guardianPhone: string | null
  createdAt: string
  updatedAt: string
  isDeleted: boolean
}

export interface StudentForm {
  id?: number
  userId?: number
  studentNo: string
  name: string
  gender: string | null
  classId: number | null
  enrollYear: string | null
  status: number
  phone: string | null
  guardianPhone: string | null
  username?: string
  password?: string
}
